aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/RealAWSCognitoAuthPlugin.kt

```diff
@@ -2223,45 +2223,56 @@ internal class RealAWSCognitoAuthPlugin(
+                                        ?.setUserMfaPreference {
+                                            this.accessToken = token
+                                            this.smsMfaSettings = sms?.let { it ->
+                                                val preferredMFASetting = it.mfaPreferred
+                                                    ?: (userPreference.preferred == MFAType.SMS && it.mfaEnabled)
+                                                SmsMfaSettingsType.invoke {
+                                                    enabled = it.mfaEnabled
+                                                    preferredMfa = preferredMFASetting
+                                                }
+                                            }
+                                            this.softwareTokenMfaSettings = totp?.let { it ->
+                                                val preferredMFASetting = it.mfaPreferred
+                                                    ?: (userPreference.preferred == MFAType.TOTP && it.mfaEnabled)
+                                                SoftwareTokenMfaSettingsType.invoke {
+                                                    enabled = it.mfaEnabled
+                                                    preferredMfa = preferredMFASetting
+                                                }
+                                            }
+                                        }?.also {
+                                            onSuccess.call()
                                         }
-                                    }
-                                }?.also {
-                                    onSuccess.call()
-                                }
-                            } ?: onError.accept(SignedOutException())
-                        } catch (error: Exception) {
-                            onError.accept(
-                                CognitoAuthExceptionConverter.lookup(
-                                    error,
-                                    "Amazon Cognito cannot update the MFA preferences"
+                                } ?: onError.accept(SignedOutException())
+                            } catch (error: Exception) {
+                                onError.accept(
+                                    CognitoAuthExceptionConverter.lookup(
+                                        error,
+                                        "Amazon Cognito cannot update the MFA preferences"
+                                    )
                                 )
-                            )
+                            }
                         }
                     }
-                }
 
-                else -> onError.accept(InvalidStateException())
+                    else -> onError.accept(InvalidStateException())
+                }
             }
-        }
+        }, {
+            onError.accept(it)
```

Comment:
We should be careful with this message being thrown. The message of failure here is if fetch fails, which will be confusing to the customer in an update call. Can we wrap the inner exception in an outer exception that has a message like "Failed to fetch current MFA preference, which is a prerequisite of updating MFA preferences.

Reply:
done